modules/dnn/src/layers/cumsum_layer.cpp

```diff
@@ -47,9 +47,30 @@ class CumSumLayerImpl CV_FINAL : public CumSumLayer
         inputs_arr.getMatVector(inputs);
         outputs_arr.getMatVector(outputs);
 
+        CV_CheckTypeEQ(inputs[0].depth(), outputs[0].depth(), "");
+
+        switch(inputs[0].depth())
+        {
+            case CV_32F:
+                forwardImpl<float>(inputs, outputs);
+                break;
+            case CV_32S:
+                forwardImpl<int32_t>(inputs, outputs);
+                break;
+            case CV_64S:
+                forwardImpl<int64_t>(inputs, outputs);
+                break;
+            default:
+                CV_Error(Error::BadDepth, "");
+        }
+    }
```

Comment:
Do we have tests for int32 and int64 data types?

Reply:
Not yet. I will add .onnx tests in opencv_extra for all the new int layers later in a separate PR